chains/solana/contracts/tests/ccip/ccip_router_test.go

```diff
@@ -2738,98 +2780,177 @@ func TestCCIPRouter(t *testing.T) {
+				require.Equal(t, 1, len(ccipMessageSentEvent.Message.TokenAmounts))
+				ta := ccipMessageSentEvent.Message.TokenAmounts[0]
+
+				require.Equal(t, wsol.mint, ccipMessageSentEvent.Message.FeeToken)
+				require.Equal(t, tokens.ToLittleEndianU256(36333028), ccipMessageSentEvent.Message.FeeTokenAmount.LeBytes)
+				// The difference is the ratio between the fee token value (wsol) and link token value (signified by token2022 in these tests).
+				// Since they have been configured in the test setup to differ by a factor of 10, so does the token amount and its value in juels
+				require.Equal(t, tokens.ToLittleEndianU256(3633302), ccipMessageSentEvent.Message.FeeValueJuels.LeBytes)
+				require.Equal(t, token0.PoolConfig, ta.SourcePoolAddress)
+				require.Equal(t, []byte{1, 2, 3}, ta.DestTokenAddress)
+				require.Equal(t, 0, len(ta.ExtraData))
+				require.Equal(t, tokens.ToLittleEndianU256(1), ta.Amount.LeBytes)
+				require.Equal(t, 32, len(ta.DestExecData))
+				expectedDestExecData := make([]byte, 32)
+				// Token0 billing had DestGasOverhead set to 100 during setup
+				binary.BigEndian.PutUint64(expectedDestExecData[24:], 100)
+				require.Equal(t, expectedDestExecData, ta.DestExecData)
+
+				// check pool event
+				poolEvent := tokens.EventBurnLock{}
+				require.NoError(t, common.ParseEvent(result.Meta.LogMessages, "Burned", &poolEvent, config.PrintEvents))
+				require.Equal(t, config.ExternalTokenPoolsSignerPDA, poolEvent.Sender)
+				require.Equal(t, uint64(1), poolEvent.Amount)
+
+				// check balances
+				_, currSupply, err := tokens.TokenSupply(ctx, solanaGoClient, token0.Mint.PublicKey(), config.DefaultCommitment)
+				require.NoError(t, err)
+				require.Equal(t, 1, initSupply-currSupply) // burned amount
+				_, currBal, err := tokens.TokenBalance(ctx, solanaGoClient, token0.User[user.PublicKey()], config.DefaultCommitment)
+				require.NoError(t, err)
+				require.Equal(t, 1, initBal-currBal) // burned amount
+				_, poolBal, err := tokens.TokenBalance(ctx, solanaGoClient, token0.PoolTokenAccount, config.DefaultCommitment)
+				require.NoError(t, err)
+				require.Equal(t, 0, poolBal) // pool burned any sent to it
+			})
+
+			t.Run("two tokens", func(t *testing.T) {
+				_, initBal0, err := tokens.TokenBalance(ctx, solanaGoClient, token0.User[user.PublicKey()], config.DefaultCommitment)
+				require.NoError(t, err)
+				_, initBal1, err := tokens.TokenBalance(ctx, solanaGoClient, token1.User[user.PublicKey()], config.DefaultCommitment)
+				require.NoError(t, err)
+
+				destinationChainSelector := config.EvmChainSelector
+				destinationChainStatePDA := config.EvmDestChainStatePDA
+				message := ccip_router.SVM2AnyMessage{
+					FeeToken: wsol.mint,
+					Receiver: validReceiverAddress[:],
+					Data:     []byte{4, 5, 6},
+					TokenAmounts: []ccip_router.SVMTokenAmount{
+						{
+							Token:  token0.Mint.PublicKey(),
+							Amount: 1,
+						},
+						{
+							Token:  token1.Mint.PublicKey(),
+							Amount: 1,
```

Comment:
nit: I generally try not to repeat values in tests, so that you can implicitly test that they are not getting swapped with no additional effort, so I'd suggest using different amounts in each token 🙂